app/src/main/java/com/keylesspalace/tusky/view/SliderPreference.kt

```diff
@@ -0,0 +1,185 @@
+package com.keylesspalace.tusky.view
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import android.view.View.VISIBLE
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.preference.Preference
+import androidx.preference.PreferenceViewHolder
+import com.google.android.material.slider.LabelFormatter.LABEL_GONE
+import com.google.android.material.slider.Slider
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.databinding.PrefSliderBinding
+import java.lang.Float.max
+import java.lang.Float.min
+
+/**
+ * Slider preference
+ *
+ * Similar to [SeekBarPreference], but better because:
+ *
+ * - Uses a [Slider] instead of a [SeekBar]. Slider supports float values, and step sizes
+ *   other than 1.
+ * - Displays the currently selected value in the Preference's summary, for consistency
+ *   with platform norms.
+ * - Icon buttons can be displayed at the start/end of the slider. Pressing them will
+ *   increment/decrement the slider by `stepSize`.
+ * - User can supply a custom formatter to format the summary value
+ */
+class SliderPreference @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.preferenceStyle,
+    defStyleRes: Int = 0
+) : Preference(context, attrs, defStyleAttr, defStyleRes),
+    Slider.OnChangeListener,
+    Slider.OnSliderTouchListener {
+
+    /** Backing property for `value` */
+    private var _value = 0F
+
+    /**
+     * @see Slider.getValue
+     * @see Slider.setValue
+     */
+    var value: Float = defaultValue
+        get() = _value
+        set(v) {
+            val clamped = max(max(v, valueFrom), min(v, valueTo))
+            if (clamped == field) return
+            _value = clamped
+            persistFloat(v)
+            notifyChanged()
+        }
+
+    /** @see Slider.setValueFrom */
+    var valueFrom: Float
+
+    /** @see Slider.setValueTo */
+    var valueTo: Float
+
+    /** @see Slider.setStepSize */
+    var stepSize: Float
+
+    /**
+     * Format string to be applied to values before setting the summary. For more control set
+     * [SliderPreference.formatter]
+     */
+    var format: String = defaultFormat
+
+    /**
+     * Function that will be used to format the summary. The default formatter formats using the
+     * value of the [SliderPreference.format] property.
+     */
+    var formatter: (Float) -> String = { format.format(it) }
+
+    /**
+     * Optional icon to show in a button at the start of the slide. If non-null the button is
+     * shown. Clicking the button decrements the value by one step.
+     */
+    var decrementIcon: Drawable? = null
+
+    /**
+     * Optional icon to show in a button at the end of the slider. If non-null the button is
+     * shown. Clicking the button increments the value by one step.
+     */
+    var incrementIcon: Drawable? = null
```

Comment:
As for the comment: Can the icon be clicked?

Reply:
Yes -- this is the icon that's shown on the button. If there's no icon then the button is hidden.